CoP Revisions

nick-mon1 opened the pull request:
This PR implements the following **changes:**

* redesigned Communities of Practice (COP) according to mockup
  * implemented card component to emphasize Digital.gov COP's
  * implemented collections component for other COP's


Design Notes:

- IT Buyers is not listed on Mockup, changed weight to 0 to reflect this
- Top paragraph could be shorter?
- Write logic to display external link icon on external links, need to do?

**URL / Link to page**

<!-- Link to the page that will be changed -->
[Digital.gov/communites](https://digital.gov/communities/)
<!-- Delete this section if not needed -->

Old COP Design:

<img width="680" alt="COP Original" src="https://user-images.githubusercontent.com/104778659/168102189-bcd7be59-c98a-4d85-9bc7-f7609ba3fbd6.png">

New Design:

<img width="697" alt="COP Digital gov Cards" src="https://user-images.githubusercontent.com/104778659/168102269-e686fa16-f4fe-4850-94d3-467d01e47195.png">

<img width="703" alt="COP Digital gov Cards with Image" src="https://user-images.githubusercontent.com/104778659/168102812-af126741-aad5-4789-bb2d-8e11b7a4a8a0.png">


<img width="743" alt="COP Digital gov Collection" src="https://user-images.githubusercontent.com/104778659/168102302-5acf285e-7370-441c-a8c5-0011b575a9eb.png">

nick-mon1 commented:
Added images for the card as per Ammie's suggestion. Will need to modify image sizes to display smaller on mobile if we decide to go with images.

<img width="665" alt="COP #3 Commit Card Desktop" src="https://user-images.githubusercontent.com/104778659/168322116-ee0d891c-0aa0-44de-8a56-c9aeb86495ab.png">
<img width="297" alt="COP #3 Commit Card Mobile" src="https://user-images.githubusercontent.com/104778659/168322120-f7f8a1a0-6ee8-4aa0-a4b3-600c57eb8b03.png">

mejiaj commented on themes/digital.gov/layouts/communities/list.html:
```diff
@@ -23,33 +23,108 @@ <h1>{{- .Title | markdownify -}} </h1>
+                  </div>
+                  <div class="usa-card__footer">
+                    <a href="{{- $e.Permalink -}}" title="{{- $e.Title | markdownify -}}">
+                      <button class="usa-button">Join Now!</button>
```
Is this **Join Now!** supposed to be a button or a link to the page?

We should also include more specific text in the call to action for accessibility.

https://accessibility.18f.gov/links/

nick-mon1 replied:
I did have a concern about wrapping buttons with links. Should just be a link to an internal page.

mejiaj reviewed (commented):
Looks good in mobile and desktop. Just added some comments/questions for _potential_ changes.

nick-mon1 commented on themes/digital.gov/layouts/communities/list.html:
```diff
@@ -23,33 +23,108 @@ <h1>{{- .Title | markdownify -}} </h1>
+                </div>
+              </li> -->
+
+              <li class="usa-card usa-card--flag tablet-lg:grid-col-12">
```
Agreed, ["Card with Media"](https://designsystem.digital.gov/components/card/) looks to be the better option.

nick-mon1 commented:
Refactored card styles with images and updated link text for accessibility.

<img width="680" alt="COP #4 Commit Card Desktop" src="https://user-images.githubusercontent.com/104778659/168376882-90accb3b-04c3-47b3-a06c-d2820e626c79.png">

nick-mon1 commented:
New image art, button link styles and updated path for search icon.
<img width="690" alt="Screen Shot 2022-05-16 at 3 37 45 PM" src="https://user-images.githubusercontent.com/104778659/168669320-51982058-4516-42dd-91b0-9e9466a7d97c.png">

nick-mon1 commented:
Link to [digital.gov/communities web archive](https://web.archive.org/web/20220516201647/https://digital.gov/communities/).

mejiaj commented on themes/digital.gov/layouts/communities/list.html:
```diff
@@ -23,33 +23,98 @@ <h1>{{- .Title | markdownify -}} </h1>
-                  <p>{{- $e.Params.summary | markdownify -}}</p>
+  <section class="usa-section">
+    <div class="grid-container-desktop">
+          <ul class="usa-collection grid-row">
```
Can we change the indents in this and other sections to the standard 2 spaces?

mejiaj commented on themes/digital.gov/layouts/communities/list.html:
```diff
@@ -23,33 +23,98 @@ <h1>{{- .Title | markdownify -}} </h1>
+  <section class="usa-section">
+    <div class="grid-container-desktop">
+          <ul class="usa-collection grid-row">
+            <div class="grid-row tablet:grid-gap-4">
```
There are divs rendering as direct children of the `ul`, let's update this markup to be `ul > li > div`

mejiaj commented on themes/digital.gov/layouts/communities/list.html:
```diff
@@ -23,33 +23,98 @@ <h1>{{- .Title | markdownify -}} </h1>
+            <div class="tablet:grid-col-6">
+              <li class="usa-collection__item">
+                <div class="usa-collection__body">
+                  <h3 class="usa-collection__heading">
```
The **Other Communities of Practice** header is an h1, so this h3 breaks the order of headers. Let's use an h2 _or_ let's change the header for the Other section from h1 to h2.

mejiaj commented on themes/digital.gov/layouts/communities/list.html:
```diff
@@ -23,33 +23,98 @@ <h1>{{- .Title | markdownify -}} </h1>
+                    <div class="usa-card__img">
+                      <img
+                        src="{{- $e.Params.dg_image | absURL -}}"
+                        alt="{{- $e.Title }}"
```
Let's use a more specific alt description for the actual images themselves instead of the title (it would read twice).

Alternatively, we could use an empty alt and add `aria-hidden="true"` for this decorative image.

nick-mon1 commented:
Replaced flexbox with grid mixin for the collections list for gutter spacing to display properly

ToniBonittoGSA commented:
Hi @nick-mon1 @mejiaj 

I checked the preview of the main CoP page https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.app.cloud.gov/preview/gsa/digitalgov.gov/cop-revised/communities/ 


1)  The IT Buyers CoP (on the [live site](https://digital.gov/communities/)) is missing from the list on the [preview](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.app.cloud.gov/preview/gsa/digitalgov.gov/cop-revised/communities/).

2) These in the `Other Communities` list are still on our site; they should not have that _open new site/window_ link icon.

API
Challenges
CX
DevOps
IT Buyers (missing)
Open Data
Open Source
OpenGov
RPA
Video Production Pros

3) The 7 at the top are repeated in the Other list; they should be removed from Other.

FCN 
Multilingual 
Plain Language 
SocialGov 
UX 
Web Analytics and Optimization 
Web Managers

nick-mon1 commented:
Fixes include:

- filtering only non digital.gov CoP pages
- added IT Buyers to the list
- hide/show logic for displaying external links list
